src/Maths/Silk.NET.Maths/Quaternion.cs

```diff
@@ -0,0 +1,771 @@
+
+                s1 = Scalar.Multiply(Scalar.Sin(Scalar.Multiply(Scalar.Subtract(Scalar<T>.One, t), omega)), invSinOmega);
+                s2 = (flip)
+                    ? Scalar.Negate(Scalar.Multiply(Scalar.Sin(Scalar.Multiply(t, omega)), invSinOmega))
+                    : Scalar.Multiply(Scalar.Sin(Scalar.Multiply(t, omega)), invSinOmega);
+            }
+
+            Quaternion<T> ans;
+
+            ans.X = Scalar.Add(Scalar.Multiply(s1, quaternion1.X), Scalar.Multiply(s2, quaternion2.X));
+            ans.Y = Scalar.Add(Scalar.Multiply(s1, quaternion1.Y), Scalar.Multiply(s2, quaternion2.Y));
+            ans.Z = Scalar.Add(Scalar.Multiply(s1, quaternion1.Z), Scalar.Multiply(s2, quaternion2.Z));
+            ans.W = Scalar.Add(Scalar.Multiply(s1, quaternion1.W), Scalar.Multiply(s2, quaternion2.W));
+
+            return ans;
+        }
+
+        /// <summary>Subtracts one Quaternion from another.</summary>
+        /// <param name="value1">The first source Quaternion.</param>
+        /// <param name="value2">The second Quaternion, to be subtracted from the first.</param>
+        /// <returns>The result of the subtraction.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Quaternion<T> Subtract(Quaternion<T> value1, Quaternion<T> value2)
+            => value1 - value2;
+
+        /// <summary>Returns a boolean indicating whether the given Object is equal to this Quaternion instance.</summary>
+        /// <param name="obj">The Object to compare against.</param>
+        /// <returns>True if the Object is equal to this Quaternion; False otherwise.</returns>
+        public override readonly bool Equals(object? obj)
+            => (obj is Quaternion<T> other) && Equals(other);
+
+        /// <summary>Returns a boolean indicating whether the given Quaternion is equal to this Quaternion instance.</summary>
+        /// <param name="other">The Quaternion to compare this instance to.</param>
+        /// <returns>True if the other Quaternion is equal to this instance; False otherwise.</returns>
+        public readonly bool Equals(Quaternion<T> other)
+            => this == other;
+
+        /// <summary>Returns the hash code for this instance.</summary>
+        /// <returns>The hash code.</returns>
+        public override readonly int GetHashCode()
+        {
+            return unchecked(X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode() + W.GetHashCode());
+        }
+
+        /// <summary>Calculates the length of the Quaternion.</summary>
+        /// <returns>The computed length of the Quaternion.</returns>
+        public readonly T Length()
+            => Scalar.Sqrt(LengthSquared());
+
+        /// <summary>Calculates the length squared of the Quaternion. This operation is cheaper than Length().</summary>
+        /// <returns>The length squared of the Quaternion.</returns>
+        public readonly T LengthSquared()
```

Comment:
These can be properties now we're going the second class serialization route

Medium priority

Reply:
I'd argue these are too expensive to be properties. Not sure.

Reply:
I still think it should be a property anyway, the only reason why i originally wanted it to be a method is because of serializers, don't see any reason why we need to discriminate by cost.

I do know what you're getting at in that a subset of users expect properties to just be there, but I don't think we should be too fussed about it. Properties are just objectively cleaner.